File: HNRZ Salary Manager/app/Models/Employee.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Employee extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'id_pekerja',
        'nik',
        'nama_lengkap',
        'no_telepon',
        'nama_bank',
        'nomor_rekening',
        'email',
        'alamat',
        'jabatan',
        'jabatan_id',
        'payroll_method_id',
        'role',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function position()
    {
        return $this->belongsTo(Jabatan::class, 'jabatan_id');
    }

    public function jabatanRelation()
    {
        return $this->position();
    }

    public function payrollMethod()
    {
        return $this->belongsTo(PayrollMethod::class, 'payroll_method_id');
    }
    public function bonuses()
    {
        return $this->belongsToMany(Bonus::class, 'bonus_employee');
    }

    public function getPositionNameAttribute(): string
    {
        $position = $this->position;

        if ($position) {
            return $position->name;
        }

        if (! empty($this->jabatan)) {
            $legacyPosition = Jabatan::query()->where('name', $this->jabatan)->first();

            if ($legacyPosition) {
                return $legacyPosition->name;
            }
        }

        return 'No Position Assigned';
    }

    public function getSalaryAttribute(): int
    {
        $position = $this->position;

        if ($position) {
            return (int) $position->salary;
        }

        if (! empty($this->jabatan)) {
            $legacyPosition = Jabatan::query()->where('name', $this->jabatan)->first();

            if ($legacyPosition) {
                return (int) $legacyPosition->salary;
            }
        }

        return 0;
    }
}

File: hnrz_salary_manager/lib/models/employee.dart
class Employee {
  final String employeeId;
  final String name;
  final String email;
  final String phone;
  final String address;
  final String position;
  final String payrollMethod;
  final bool isEmployee;
  final int baseSalary;

  const Employee({
    required this.employeeId,
    required this.name,
    required this.email,
    required this.phone,
    required this.address,
    required this.position,
    required this.payrollMethod,
    required this.isEmployee,
    required this.baseSalary,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    final position = json['position'] as Map<String, dynamic>?;
    final payrollMethod = json['payroll_method'] as Map<String, dynamic>?;

    return Employee(
      employeeId: json['id_pekerja']?.toString() ?? '-',
      name: json['nama_lengkap']?.toString() ?? '-',
      email: json['email']?.toString() ?? '-',
      phone: json['no_telepon']?.toString() ?? '-',
      address: json['alamat']?.toString() ?? '-',
      position: position?['name']?.toString() ?? json['jabatan']?.toString() ?? '-',
      payrollMethod: payrollMethod?['name']?.toString() ?? '-',
      isEmployee: json['is_employee'] == true,
      baseSalary: (position?['salary'] as num?)?.toInt() ?? 0,
    );
  }
}
